v0/destinations/delighted/transform.js

```diff
@@ -0,0 +1,238 @@
+  if (!userId) {
+    throw new CustomError("userId is required for identify", 400);
+  }
+  let channel =
+    getDestinationExternalID(message, "delightedChannelType") || Config.channel;
+  channel = channel.toLowerCase();
+
+  isValidUserIdOrError(channel, userId);
+  let payload = constructPayload(message, identifyMapping);
+  
+  if(channel === "email"){
+    payload.email = userId;
+    if(payload.phone_number && !isValidPhone(payload.phone_number)){
+      payload.phone_number= null;
+      logger.error("Phone number format must be E.164.");
+  }
+  }else if(channel == "phone"){
+    payload.phone_number = userId;
+    if(payload.email && !isValidEmail(payload.email)){
+      payload.email = null;
+      logger.error("Email format is not correct.");
+    }
+  }
+  
+  payload.send = "false";
+  payload.channel = channel;
+  payload.delay = Config.delay || 0;
+  if (!payload.name) {
+    const fName = getFieldValueFromMessage(message, "firstName").trim();
+    const lName = getFieldValueFromMessage(message, "lastName").trim();
+    const name = fName.concat(" ").concat(lName);
+    if (name) {
+      payload.name = name;
+    }
+  }
+  let properties = {};
+  properties = extractCustomFields(
+    message,
+    properties,
+    ["traits", "context.traits"],
+    DELIGHTED_EXCLUSION_FIELDS
+  );
+  if(!isEmptyObject(properties)){
+    payload = {
+      ...payload,
+      properties
+    };
+  }
+  let lastSentAt= getValueFromMessage(message,"lastSentAt");
+  if(lastSentAt !== undefined && lastSentAt !== null)
+  payload.last_sent_at= lastSentAt;
+
+  // update/create the user
+  const response = defaultRequestConfig();
+  response.headers = {
+    Authorization: `Basic ${Config.apiKey}`,
+    "Content-Type": "application/json"
+  };
+  response.method = defaultPostRequestConfig.requestMethod;
+  response.endpoint = ENDPOINT;
+  response.body.JSON = removeUndefinedAndNullValues(payload);
+  return response;
+};
+
+const trackResponseBuilder = async (message, { Config }) => {
+  // checks if the event is valid if not throws error else nothing
+  eventValidity(Config, message);
+
+  const userId = getFieldValueFromMessage(message, "userIdOnly");
+  if (!userId) {
+    throw new CustomError("userId is required.", 400);
+  }
+  let channel =
+    getDestinationExternalID(message, "delightedChannelType") || Config.channel;
+  channel = channel.toLowerCase();
+
+  isValidUserIdOrError(channel, userId);
+
+  // checking if user already exists or not, throw error if it doesn't
+  const check = await userValidity(channel, Config, userId);
+  if (!check) {
+    throw new CustomError(`user ${userId} doesnot exist`, 400);
+  }
+  const payload = {};
+
+  payload.send = "true";
+  payload.channel = channel;
+  payload.delay = Config.delay || message.properties.delay || 0;
+  if(message.properties.lastSentAt){
+  payload.last_sent_at = message.properties.lastSentAt;
+  }
+
+  if (message.properties) payload.properties = message.properties;
```

Comment:
this check should be done, before `if(message.properties.lastSentAt)`
also if `properties` has lastSentAt, lets delete it from there, otherwise it will be sent as a custom property as well

Reply:
yeah updating